### ISSUE TITLE
Build failure: clisp on x86_64-darwin in preConfigure

### ISSUE THREAD

**Issue by tianyaochou** [NixOS/nixpkgs#258050]
### Steps To Reproduce

1. build *clisp*

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/b3s7xxpgwwwxcfdnyrnir3v75fagz6sp-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/1hh91wzm5hy44fmv6faypf810jhdgzy1-gnulib_aarch64.patch
patching file src/gllib/vma-iter.c
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
autoreconf: configure.in: AM_GNU_GETTEXT is used, but not AM_GNU_GETTEXT_VERSION
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
libtoolize: copying file 'build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
libtoolize: Consider adding '-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
automake: warning: autoconf input should be named 'configure.ac', not 'configure.in'
configure.in:34: installing 'build-aux/compile'
configure.in:23: installing 'build-aux/missing'
automake: warning: autoconf input should be named 'configure.ac', not 'configure.in'
gllib/Makefile.am: installing 'build-aux/depcomp'
/private/tmp/nix-build-clisp-2.50pre20230112.drv-0/source
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
autoreconf: cannot create $cl_cv_clisp_libdir/build-aux: No such file or directory
/private/tmp/nix-build-clisp-2.50pre20230112.drv-0/source
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
autoreconf: cannot create $cl_cv_clisp_libdir/build-aux: No such file or directory
/private/tmp/nix-build-clisp-2.50pre20230112.drv-0/source
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
aclocal: warning: couldn't open directory '../../src': No such file or directory
aclocal: error: couldn't open directory '../../src/m4': No such file or directory
autoreconf: aclocal failed with exit status: 1
/nix/store/rz54rq2civz8a7hlqgw30japnycv4d0q-stdenv-darwin/setup: line 153: pop_var_context: head of shell_variables not a function context
```

### Additional context

Fail to build clisp on x86_64-darwin

### Notify maintainers

@raskin, @tobim, @lukego, @nagy, @uthar, @hraban

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix run nixpkgs#nix-info -- -m
 - system: `"x86_64-darwin"`
 - host os: `Darwin 23.0.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.15.1`
 - channels(tianyaochou): `""`
 - channels(root): `"nixpkgs-22.05-darwin-22.05-darwin"`
 - nixpkgs: `not found`
```

**Comment by hraban** [comment]
For context: afaik this was previously marked broken on all darwin systems, so this isn't a regression so much as "we could choose to mark x86_64-darwin broken again". It does work on aarch64-darwin.

**Comment by tianyaochou** [comment]
Sorry, I wasn't clear. It fails at `preConfigure` stage, where the derivation configures modules. I did succeed to build it with
```
unpackPhase
cd source
patchPhase
./configure --with-readline=true
cd src
make
```

Is there any difference in the way to build the package in nixpkgs from this?

**Comment by tobim** [comment]
Sorry for removing the broken flag on x86_64 when I only tested aarch64.

**Comment by hraban** [comment]
This is one of those tooling issues IMO. I'm surprised we are allowed to merge PRs with broken builds like this without much noise from CI.

**Comment by hraban** [comment]
I've been taking a look at this and I think there are two problems:

1. the configure phase is failing early in nix build, but not when you run it yourself manually (try `nix develop github:nixos/nixpkgs/nixpkgs-unstable#clisp && unpackPhase && cd source && patchPhase && configurePhase && buildPhase`--it will get much further than a standard `nix build ...#clisp`). I'm not sure what the problem is here.
2. Once you get past configure, actual build of clisp is still broken with all the currently enabled flags on x86_64-darwin.

I have a fix for issue 1, which is cosmetic, but at least it makes the build in nixpkgs "fail consistently". :)

**Comment by hraban** [comment]
Fixed in #258073.

**Comment by tianyaochou** [comment]
I can confirm #258073 fixes it. Thanks everyone!
